frontend/src/pages/NewStudent.vue

```diff
@@ -132,37 +121,54 @@ export default {
     }
   },
 
-  computed: {
-    filteredLanguageDataArray() {
-        return this.languageData.filter((option) => {
-            return option
-              .toString()
-              .toLowerCase()
-              .indexOf(this.nativeLanguage.toLowerCase()) >= 0
-            })
-    },
-      filteredSourceDataArray() {
-        return this.sourceData.filter((sourceOption) => {
-            return sourceOption
-              .toString()
-              .toLowerCase()
-              .indexOf(this.source.toLowerCase()) >= 0
-            })
-    }  
+ async mounted() {
+    // Call API for Native Languages
+      await fetch("/api/nativeLanguages")
+        .then(response => response.json())
+        .then(result => this.API_nativeLanguage = result)
   },
 
-
   methods: {
     createStudent(){
+       const studentCreate = {
+        method: "POST",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          PhoneNumber: this.PhoneNumber,
+          FirstName: this.name,
+          LastName: "Placeholder",
+          //FullName: this.name,
+          Source: this.source,
+          NativeLanguageID: this.selected.NativeLanguageID, // how to avoid hardcoding the languageID? can the backend process text then match to the appropriate ID? 
+          EnglishProficiency: this.EnglishProficiency, // must be No, Little, Simple or Intermediate. form input is int 
+          Notes: this.Notes,
+          StatusString: "SCREENING"
+        })
+      }
+      fetch("/api/students", studentCreate)
+        .then(response => response.json()) 
+      //Pop-up notification that new student has been added
       this.$buefy.notification.open({
         message: 'New student added. <u>View profile</u>!',
         duration: 5000,
         type: 'is-success',
         position: 'is-top',
         // color: '#57A773',
       })
+      setTimeout(() => {this.$router.push({path: `/students`})}, );
```

Comment:
You should apply the same approach here to what I mentioned for the notification (only doing it after the successful creation).

Reply:
latest commit addresses this issue by disabling the Create Student button unless Name, Phone Number and NativeLanguage are present